src/View/WebView.php

```diff
@@ -79,6 +85,14 @@ class WebView extends View
      */
     private const PH_BODY_END = '<![CDATA[YII-BLOCK-BODY-END]]>';
 
+    /**
+     * The widgets that are currently being rendered (not ended). This property is maintained by {@see beginWidget()}
+     * and {@see endWidget()} methods.
+     *
+     * @var Widget[] $stack
+     */
+    protected $stack;
```

Comment:
Why not private?

Reply:
I thought that someone might need to redefine the behavior of the stack, for example, add additional checks for emptiness.